src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -5116,6 +5116,368 @@ .maxstack  2
+    {
+        Console.Write(M0("""", 0)); // 0
+        Console.Write(M0("""", 1)); // 1
+        Console.Write(M0("""", 2)); // 2
+        Console.Write(M0("""", 3)); // 3
+        Console.Write(M0(""a"", 2)); // 2
+        Console.Write(M0(""a"", 10)); // 3
+    }
+
+    static int M0(string x, int y)
+    {
+        return (x, y) switch
+        {
+            ("""", 0) => M1(0),
+            ("""", 1) => M1(1),
+            (_, 2) => M1(2),
+            _ => M1(3)
+        };
+    }
+
+    static int M1(int z)
+    {
+        Console.Write(' ');
+        return z;
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: " 0 1 2 3 2 3", options: TestOptions.DebugExe);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("Program.M0", @"{
+  // Code size       90 (0x5a)
+  .maxstack  2
+  .locals init (int V_0,
+                int V_1)
+  IL_0000:  nop
+  IL_0001:  ldc.i4.1
+  IL_0002:  brtrue.s   IL_0005
+  IL_0004:  nop
+  IL_0005:  ldarg.0
+  IL_0006:  ldstr      """"
+  IL_000b:  call       ""bool string.op_Equality(string, string)""
+  IL_0010:  brfalse.s  IL_0026
+  IL_0012:  ldarg.1
+  IL_0013:  switch    (
+        IL_002c,
+        IL_0035,
+        IL_003e)
+  IL_0024:  br.s       IL_0047
+  IL_0026:  ldarg.1
+  IL_0027:  ldc.i4.2
+  IL_0028:  beq.s      IL_003e
+  IL_002a:  br.s       IL_0047
+  IL_002c:  ldc.i4.0
+  IL_002d:  call       ""int Program.M1(int)""
+  IL_0032:  stloc.0
+  IL_0033:  br.s       IL_0050
+  IL_0035:  ldc.i4.1
+  IL_0036:  call       ""int Program.M1(int)""
+  IL_003b:  stloc.0
+  IL_003c:  br.s       IL_0050
+  IL_003e:  ldc.i4.2
+  IL_003f:  call       ""int Program.M1(int)""
+  IL_0044:  stloc.0
+  IL_0045:  br.s       IL_0050
+  IL_0047:  ldc.i4.3
+  IL_0048:  call       ""int Program.M1(int)""
+  IL_004d:  stloc.0
+  IL_004e:  br.s       IL_0050
+  IL_0050:  ldc.i4.1
+  IL_0051:  brtrue.s   IL_0054
+  IL_0053:  nop
+  IL_0054:  ldloc.0
+  IL_0055:  stloc.1
+  IL_0056:  br.s       IL_0058
+  IL_0058:  ldloc.1
+  IL_0059:  ret
+}");
+        }
+
+        [Fact, WorkItem(46536, "https://github.com/dotnet/roslyn/issues/46536")]
+        public void MultiplePathsToNode_SwitchDispatch_03()
+        {
+            var source = @"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        Console.Write(M("""", 0)); // 0
+        Console.Write(M("""", 1)); // 1
+        Console.Write(M("""", 2)); // 2
+        Console.Write(M("""", 3)); // 3
+        Console.Write(M("""", 4)); // 4
+        Console.Write(M("""", 5)); // 5
+        Console.Write(M(""a"", 2)); // 2
+        Console.Write(M(""a"", 3)); // 3
+        Console.Write(M(""a"", 4)); // 4
+        Console.Write(M(""a"", 10)); // 5
+    }
+
+    static int M(string x, int y)
+    {
+        return (x, y) switch
+        {
+            ("""", 0) => 0,
+            ("""", 1) => 1,
+            (_, 2) => 2,
+            (_, 3) => 3,
+            (_, 4) => 4,
+            _ => 5
```

Comment:
I think this is a scenario where we'd want a second `switch` to be generated (one for the case where `x` is `""`, and one for the other cases). Can you file an issue on that?

Reply:
Opened #47145